source/extensions/tracers/zipkin/zipkin_tracer_impl.cc

```diff
@@ -68,10 +68,14 @@ Driver::Driver(const Json::Object& config, Upstream::ClusterManager& cluster_man
   const std::string collector_endpoint =
       config.getString("collector_endpoint", ZipkinCoreConstants::get().DEFAULT_COLLECTOR_ENDPOINT);
 
-  tls_->set([this, collector_endpoint, &random_generator](
+  const bool trace_id_128bit =
```

Comment:
This is always going to return false. You can just remove these lines.

Reply:
I'm a bit confused, as when I test with the jaegertracing example updating the front-proxy yaml to include the 'trace_id_128bit: true', it uses a 128bit trace id - so this must mean that the value in the yaml is being converted to json and accessed via this condition?

If that is not how the v2 config is working, is there some docs that explain how the v2 and v1 configs are being used?

Reply:
From further examination of the code it looks like many components have a factory that can take the v1 json or v2 proto message - however the tracer extension doesn't seem to have been updated to support the v2 proto config directly - so assume it is json conversion from the proto message? If so, then may be better to leave this in for now until proper separation between the v1 and v2 config for the tracer extension.

Reply:
v2 json isn't exactly the same as v2 proto (for example in proto we use byte arrays for IDs)

https://github.com/openzipkin/zipkin-api/blob/master/zipkin2-api.yaml#L298
https://github.com/openzipkin/zipkin-api/blob/master/zipkin.proto

Note the proto format is quite new, so requires a very recent version of zipkin-server or clone

Reply:
@adriancole Sorry Adrian, I wasn't clear - I am referring to the envoy v1/v2 APIs.

Reply:
@objectiser hmm. Sorry. I guess we don't have direct proto factory functions for the tracers yet. Sorry, I thought we had already added this. This will need to be added. I will open a follow up issue for this. Yes don't worry about it for now.